include/spark_dsg/scene_graph_layer.h

```diff
@@ -155,45 +224,35 @@ class SceneGraphLayer : public BaseLayer {
    */
   bool rewireEdge(NodeId source, NodeId target, NodeId new_source, NodeId new_target);
 
+  /**
+   * @brief Add the other layer into this one
+   * @param other Layer to merge into this layer
+   * @param config Merge configuration controlling contraction and attributes
+   * @param new_nodes Optional output to register new nodes
+   */
   void mergeLayer(const SceneGraphLayer& other,
                   const GraphMergeConfig& config,
                   std::vector<NodeId>* new_nodes);
 
-  /**
-   * @brief Number of nodes in the layer
-   */
-  inline size_t numNodes() const { return nodes_.size(); }
-
-  /**
-   * @brief Number of edges in the layer
-   */
-  inline size_t numEdges() const { return edges_.size(); }
-
   /**
    * @brief Get node ids of newly inserted nodes
    */
-  void getNewNodes(std::vector<NodeId>& new_nodes, bool clear_new) override;
-
-  /**
-   * @brief Get node id of deleted nodes
-   */
-  void getRemovedNodes(std::vector<NodeId>& removed_nodes, bool clear_removed) override;
+  void getNewNodes(std::vector<NodeId>& new_nodes, bool clear_new) const;
```

Comment:
I agree, but these get used in the scene graph class to populate all the new nodes, etc. for every layer (so it feels slightly cleaner to fill an existing vector instead of returning a new one and inserting)